coordinator/src/main/scala/com/socrata/datacoordinator/resources/collocation/CollocationSodaResource.scala

```diff
@@ -72,4 +70,14 @@ abstract class CollocationSodaResource extends BasicSodaResource {
         errorResponse(BadRequest, BodyRequestError.MALFORMED_JSON, "message" -> JString(e.message))
     }
   }
+
+  def withJobId(jobId: String, req: HttpRequest)(handleRequest: UUID => HttpResponse): HttpResponse = {
+    try {
+      val job = UUID.fromString(jobId)
+      handleRequest(job)
+    } catch {
+      case error: IllegalArgumentException =>
```

Comment:
It is hard to guarantee that IllegalArgumentException is not thrown in handleRequest (vs UUID.fromString).  Maybe handleRequest should be moved outside the try block?